### DISCUSSION TITLE
"ignore" a relation

### DISCUSSION THREAD

**Discussion by Rechdan** [prisma/prisma#2709] [Q&A] [answered]
So, my question is pretty straightforward, how should I ignore this relation?

**The relation is inside that red square:**

![img](https://image.prntscr.com/image/kX0MF2pNTHKO6-HF5qu-sw.png)

**Prisma code:**

```prisma
generator client {
   provider = "prisma-client-js"
}

datasource db {
   provider = "mysql"
   url      = env("DATABASE_URL")
}

model User {
   @@map(name: "user")

   id      Int      @default(autoincrement()) @id
   email   String   @unique
   created DateTime @default(now())

   friends         Friend[] @relation(name: "friend_user_id")
   friends_ignored Friend[] @relation(name: "friend_friend_id")

   @@index([created])
}

model Friend {
   @@map(name: "friend")

   id        Int      @default(autoincrement()) @id
   user_id   Int
   friend_id Int
   created   DateTime @default(now())

   user   User @relation(name: "friend_user_id", fields: [user_id], references: [id])
   friend User @relation(name: "friend_friend_id", fields: [friend_id], references: [id])

   @@index([created])
   @@unique([user_id, friend_id])
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @Rechdan :wave: 

Could you explain a bit about what ignoring the field means in this case?

**Reply by Rechdan** [reply]
I meant to ignore this field in the user's table but keep the relation between the user's table and the friend's table.

**Reply by ryands17** [reply]
This is not possible as `prisma introspect` will take into account all the tables and relations present so it is not possible at the moment to ignore a specific field/relation. Could you create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can look into it further?

**Reply by Rechdan** [reply]
I will do that, thanks for the tip!

**Reply by jackbravo** [reply]
@Rechdan, was the issue created?

I want to ignore some tables, instead of some fields.

I'm using sqlite3 ft5  virtual tables, but they create shadow tables that I would like to ignore.

**Reply by Rechdan** [reply]
@jackbravo it was not, I decided to let the relations and move forward.

My database setup now is I do the migration with TypeORM and then introspect with Prisma, it is working extremely good!

**Reply by tomhoule** [reply]
Note that there is also an `@ignore` attribute you can add to relations so they aren't used in the generated client.

**Comment by ludralph** [comment]
Hi @Rechdan,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
